packages/kbn-pm/src/utils/bazel/setup_remote_cache.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import dedent from 'dedent';
+import { existsSync, writeFileSync } from 'fs';
+import { resolve } from 'path';
+import { spawn } from '../child_process';
+import { log } from '../log';
+
+async function isVaultAvailable() {
+  try {
+    await spawn('vault', ['--version'], { stdio: 'pipe' });
+
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+async function isElasticCommitter() {
+  try {
+    const { stdout: email } = await spawn('git', ['config', 'user.email'], {
+      stdio: 'pipe',
+    });
+
+    return email.trim().endsWith('@elastic.co');
+  } catch {
+    return false;
+  }
+}
+
+export async function setupRemoteCache(repoRootPath: string) {
+  // The remote cache is only for Elastic employees working locally (CI cache settings are handled elsewhere)
+  if (process.env.CI || !(await isElasticCommitter())) {
+    return;
+  }
+
+  log.debug(`[bazel_tools] setting up remote cache settings if necessary`);
+
+  const settingsPath = resolve(repoRootPath, '.bazelrc.cache');
+
+  if (existsSync(settingsPath)) {
+    log.debug(`[bazel_tools] remote cache settings already exist, skipping`);
```

Comment:
We may need a mechanism to update the API key at some point.  Not a big deal for this, can probably do it at runtime.

Reply:
Yeah, I was thinking if we ever need that later, we can just add it. For now, you can just delete the file and get a new key.

That's actually why I added `# V1` as a comment in the file. We can check that and re-generate it if we need a `V2` later